### PR TITLE
skip when utmp is missing

### PR DESCRIPTION
**Pull request by ehlerst**
### Required for all PRs:

- [x ] Signed [CLA](https://influxdata.com/community/cla/).
- [x ] Associated README.md updated.
- [ ] Has appropriate unit tests.

Replacement for: https://github.com/influxdata/telegraf/pull/5734

### CODE REVIEW

**Comment by ehlerst** [comment]
@danielnelson is this kind of what you proposed? I took inspiration from the perf metrics plugins warning log entries.

Should the warning be enabled by default? I have no preference here.

File: plugins/inputs/system/system.go [hunk 1]
```diff
@@ -44,7 +49,9 @@ func (_ *SystemStats) Gather(acc telegraf.Accumulator) error {
 	users, err := host.Users()
 	if err == nil {
 		fields["n_users"] = len(users)
-	} else if !os.IsPermission(err) {
+	} else if os.IsNotExist(err) && s.WarnOnMissing {
+		log.Printf("[inputs.system]: %s\n", err.Error())
```
Comment (danielnelson) [inline plugins/inputs/system/system.go]: Let's do debug level:
```
log.Printf("D! [inputs.system] Error reading users: %v, err")
```

File: plugins/inputs/system/system.go [hunk 2]
```diff
@@ -25,10 +28,12 @@ func (_ *SystemStats) SampleConfig() string {
 	return `
   ## Uncomment to remove deprecated metrics.
   # fielddrop = ["uptime_format"]
+  ## Print out when utmp is not available for n_users
+  # warn_on_missing = true
```
Comment (danielnelson) [inline plugins/inputs/system/system.go]: I think we don't need an option, let's just do it unconditionally but at a level that won't be too annoying.

Comment (ehlerst) [reply plugins/inputs/system/system.go]: Removed

File: plugins/inputs/system/system.go (hunk 1 above)
Comment (ehlerst) [reply plugins/inputs/system/system.go]: updated

**Comment by ehlerst** [comment]
I think this may be gtg, i'm not sure how to make a Unit test for this since its calling a system file we probably don't want to fiddle with. Let me know.

**Pull request merged**